### PR TITLE
WT-10555 Implement import GCP and Azure SDK via package

### PR DESCRIPTION
**Pull request by jiechenbo**
This ticket aims to implement the package option for when IMPORT_GCP_SDK or IMPORT_AZURE_SDK flag is set to `package`. The package option assumes that the SDK is installed on the system and can be found via find_package().

### CODE REVIEW

File: ext/storage_sources/gcp_store/README.md [hunk 1]
```diff
@@ -16,18 +16,21 @@ This section describes how to build WiredTiger with the GCP extension enabled.
 * clang 6.0.0 or higher
 
 ### Building
-There is currently only 1 way to build WiredTiger with the GCP extension:
-This way manages the GCP SDK dependency as an external project. This method will download the
-GCP SDK, link to WiredTiger's build system, and build the extension.
+There are two ways to build WiredTiger with the GCP extension:
+
+1. Using a system installation of the GCP SDK. Information about how to perform a system install 
+from the SDK can be found [here](https://github.com/googleapis/google-cloud-cpp/blob/main/doc/packaging.md.).
```
Comment (korteland) [inline ext/storage_sources/gcp_store/README.md:22]: Grammar: "from the SDK" -> "to the SDK"

File: ext/storage_sources/azure_store/README.md [hunk 2]
```diff
@@ -13,18 +13,20 @@ This section describes how to build WiredTiger with the Azure extension enabled.
...
 * `ENABLE_AZURE=1` is required to build the Azure extension.
-* `IMPORT_AZURE_SDK={external}` is used to set the build method.
-    *   `external` tells the compiler to search for an existing system installation of the SDK.
+* `IMPORT_AZURE_SDK={external,package}` is used to set the build method.
+    *   `external` tells the compiler to download and install the SDK as part of the build.
+    *   `package` tells the compiler to search for an existing system installation of the SDK.
     *    This flag should be set alongside the `ENABLE_AZURE` flag.
-    *    If the `IMPORT_AZURE_SDK` flag is not specified, the compiler will assume a system
-            installation of the SDK which will be implemented at a later stage.
+    *    If the `IMPORT_AZURE_SDK` flag is not specified,  the compiler will assume a system
```
Comment (korteland) [inline ext/storage_sources/azure_store/README.md:28]: Nitpick: there are now two spaces between "specified," and "compiler".

File: ext/storage_sources/azure_store/CMakeLists.txt [hunk 3]
```diff
@@ -27,12 +27,12 @@ target_compile_options(
...
-        azure_core_lib
-        azure_storage_lib
-        azure_storage_common_lib
+        Azure::azure-core
+        Azure::azure-storage-blobs
+        Azure::azure-storage-common
 )
 
-add_dependencies(azure_container_util azure-sdk)
+#add_dependencies(azure_container_util azure-sdk)
```
Comment (korteland) [inline ext/storage_sources/azure_store/CMakeLists.txt:35]: Why was this commented out?

**Review by korteland — COMMENTED** [review]
Nice work on the CMake changes, they look good - just a couple of fiddly things to fix.

File: ext/storage_sources/azure_store/CMakeLists.txt (hunk 3 above)
Comment (jiechenbo) [reply ext/storage_sources/azure_store/CMakeLists.txt]: [3433d6a](https://github.com/wiredtiger/wiredtiger/pull/9000/commits/3433d6a7002b208627e313092587c0b6cf019f25)

File: ext/storage_sources/azure_store/README.md (hunk 2 above)
Comment (jiechenbo) [reply ext/storage_sources/azure_store/README.md]: [3433d6a](https://github.com/wiredtiger/wiredtiger/pull/9000/commits/3433d6a7002b208627e313092587c0b6cf019f25)

File: ext/storage_sources/gcp_store/README.md (hunk 1 above)
Comment (jiechenbo) [reply ext/storage_sources/gcp_store/README.md]: [3433d6a](https://github.com/wiredtiger/wiredtiger/pull/9000/commits/3433d6a7002b208627e313092587c0b6cf019f25)

File: ext/storage_sources/azure_store/CMakeLists.txt (hunk 3 above)
Comment (jiechenbo) [reply ext/storage_sources/azure_store/CMakeLists.txt]: I was meant to remove this, the add dependencies is added within the azure_sdk.cmake file instead!

**Review by korteland — APPROVED** [review]
Thanks, this LGTM now.

File: ext/storage_sources/azure_store/README.md [hunk 4]
```diff
@@ -13,18 +13,20 @@ This section describes how to build WiredTiger with the Azure extension enabled.
 * clang version 6.0.0 or higher
 
 ### Building
+There are two ways to build WiredTiger with the Azure extension:
 
-There is currently only 1 way to build WiredTiger with the Azure extension:
-This way manages the Azure SDK dependency as an external project. This method will download the
-Azure SDK, link to WiredTiger's build system, and build the extension.
+1. Using a system installation of the Azure SDK.
+2. Letting CMake manage the Azure SDK as an external project, this method will download the SDK,
```
Comment (tammybailey) [inline ext/storage_sources/azure_store/README.md:19]: `project, this` -> `project. This`

It was previously correct :)

File: ext/storage_sources/gcp_store/README.md [hunk 5]
```diff
@@ -16,18 +16,21 @@ This section describes how to build WiredTiger with the GCP extension enabled.
...
 
 ### Building
-There is currently only 1 way to build WiredTiger with the GCP extension:
-This way manages the GCP SDK dependency as an external project. This method will download the
-GCP SDK, link to WiredTiger's build system, and build the extension.
+There are two ways to build WiredTiger with the GCP extension:
+
+1. Using a system installation of the GCP SDK. Information about how to perform a system install 
+to the SDK can be found [here](https://github.com/googleapis/google-cloud-cpp/blob/main/doc/packaging.md.).
+2. Letting CMake manage the GCP SDK as an external project, this method will download the SDK,
```
Comment (tammybailey) [inline ext/storage_sources/gcp_store/README.md:23]: `project, this` -> `project. This`

**Review by tammybailey — APPROVED** [review]
Two grammar changes please! Otherwise LGTM.

**Comment by jiechenbo** [comment]
evergreen merge

**Pull request merged**